### PR TITLE
TRUS-2389 Remove a trustee

### PR DESCRIPTION
**Pull request by adamconder**

### CODE REVIEW

File: app/forms/YesNoFormProvider.scala~HEAD [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (JoshuaH01) [inline app/forms/YesNoFormProvider.scala~HEAD]: FIle name change

File: app/navigation/leadtrustee/IndividualLeadTrusteeNavigator.scala [hunk 2]
```diff
@@ -65,4 +73,39 @@ object IndividualLeadTrusteeNavigator {
               .map(if (_) yesCall else noCall)
               .getOrElse(controllers.routes.SessionExpiredController.onPageLoad())
   }
+
```
Comment (JoshuaH01) [inline app/navigation/leadtrustee/IndividualLeadTrusteeNavigator.scala]: Remove IndividualLeadTrusteeNav file

File: app/queries/RemoveTrusteeQuery.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
```
Comment (JoshuaH01) [inline app/queries/RemoveTrusteeQuery.scala]: Delete RemoveTrusteeQuery file

File: app/viewmodels/addAnother/LeadTrusteeIndividualViewModel.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (JoshuaH01) [inline app/viewmodels/addAnother/LeadTrusteeIndividualViewModel.scala]: Remove LeadTrusteeIndividualViewModel

File: app/viewmodels/addAnother/TrusteeViewModel.scala [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
```
Comment (JoshuaH01) [inline app/viewmodels/addAnother/TrusteeViewModel.scala]: Remove TrusteeViewModel

File: conf/app.routes [hunk 6]
```diff
@@ -21,6 +21,16 @@ GET        /trust-not-claimed                                               cont
 
 GET        /agent-not-authorised                                            controllers.AgentNotAuthorisedController.onPageLoad()
 
+GET        /add                                                             controllers.AddATrusteeController.onPageLoad()
+POST       /add                                                             controllers.AddATrusteeController.submitOne()
+POST       /add-another                                                     controllers.AddATrusteeController.submitAnother()
+
+GET        /trustee/:index/remove                                           controllers.trustee.RemoveIndividualTrusteeController.onPageLoad(index : Int)
+POST       /trustee/:index/remove                                           controllers.trustee.RemoveIndividualTrusteeController.onSubmit(index : Int)
+
```
Comment (JoshuaH01) [inline conf/app.routes]: Remove lines 30-33 lead-trustee route

File: app/forms/YesNoFormProvider.scala~HEAD (hunk 1 above)
Comment (jassalrichy) [reply app/forms/YesNoFormProvider.scala~HEAD]: Seems to have resolved itself.

File: app/navigation/leadtrustee/IndividualLeadTrusteeNavigator.scala (hunk 2 above)
Comment (jassalrichy) [reply app/navigation/leadtrustee/IndividualLeadTrusteeNavigator.scala]: Need to keep.

File: app/queries/RemoveTrusteeQuery.scala (hunk 3 above)
Comment (jassalrichy) [reply app/queries/RemoveTrusteeQuery.scala]: Done.

File: app/viewmodels/addAnother/LeadTrusteeIndividualViewModel.scala (hunk 4 above)
Comment (jassalrichy) [reply app/viewmodels/addAnother/LeadTrusteeIndividualViewModel.scala]: Done.

File: app/viewmodels/addAnother/TrusteeViewModel.scala (hunk 5 above)
Comment (jassalrichy) [reply app/viewmodels/addAnother/TrusteeViewModel.scala]: Done.

File: conf/app.routes (hunk 6 above)
Comment (jassalrichy) [reply conf/app.routes]: Done

File: app/forms/DateRemovedFromTrustFormProvider.scala [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+class DateRemovedFromTrustFormProvider @Inject() extends Mappings {
+
+  def withPrefixAndTrustStartDate(prefix: String, trustStartDate: LocalDate): Form[LocalDate] =
+    Form(
+      "value" -> localDate(
+        invalidKey     = s"$prefix.error.invalid",
+        allRequiredKey = s"$prefix.error.required.all",
+        twoRequiredKey = s"$prefix.error.required.two",
+        requiredKey    = s"$prefix.error.required"
+      ).verifying(firstError(
```
Comment (adamconder) [inline app/forms/DateRemovedFromTrustFormProvider.scala]: Have a betweenDates constraint for the form providers.

File: app/models/Name.scala [hunk 8]
```diff
@@ -20,8 +20,9 @@ import play.api.libs.json._
 
 case class Name (firstName: String, middleName: Option[String], lastName: String) {
   lazy val displayName : String = firstName + " " + lastName
+  override def toString = s"$firstName $lastName"
```
Comment (adamconder) [inline app/models/Name.scala]: Remove the toString method.

File: app/models/RemoveTrusteeIndividual.scala [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package models
+
+import org.joda.time.DateTime
+import play.api.libs.json.{Format, Json, Reads, Writes, __}
+import play.api.libs.functional.syntax._
+
+case class RemoveTrusteeIndividual(lineNo: Option[String],
```
Comment (adamconder) [inline app/models/RemoveTrusteeIndividual.scala]: Replace this with Individual Trustee model

File: app/models/RemoveTrusteeOrg.scala [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+package models
+
+import java.time.LocalDate
+
+import org.joda.time.DateTime
+import play.api.libs.json.{Format, Json, Reads, Writes}
+
+case class RemoveTrusteeOrg(lineNo: Option[String],
```
Comment (adamconder) [inline app/models/RemoveTrusteeOrg.scala]: See above.

File: app/models/TrusteeStatus.scala [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+sealed trait TrusteeStatus
+
+object TrusteeStatus extends Enumerable.Implicits {
```
Comment (adamconder) [inline app/models/TrusteeStatus.scala]: Delete this file

File: app/models/Trustees.scala [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json.{Format, Json}
+
+case class TrusteeType(
```
Comment (adamconder) [inline app/models/Trustees.scala]: This won't be needed when moving to the TrusteeType

File: app/pages/leadtrustee/individual/LeadTrusteeStatusPage.scala [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package pages.leadtrustee.individual
+
+import models.TrusteeStatus
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object LeadTrusteeStatusPage extends QuestionPage[TrusteeStatus] {
```
Comment (adamconder) [inline app/pages/leadtrustee/individual/LeadTrusteeStatusPage.scala]: Remove this

File: app/pages/trustee/AddATrusteePage.scala [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package pages.trustee
+
+import models.AddATrustee
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object AddATrusteePage extends QuestionPage[AddATrustee] {
```
Comment (adamconder) [inline app/pages/trustee/AddATrusteePage.scala]: Remove this.

File: app/pages/trustee/AddATrusteeYesNoPage.scala [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.trustee
+
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object AddATrusteeYesNoPage extends QuestionPage[Boolean] {
```
Comment (adamconder) [inline app/pages/trustee/AddATrusteeYesNoPage.scala]: Remove this.

File: app/pages/trustee/IsThisLeadTrusteePage.scala [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.trustee
+
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+final case class IsThisLeadTrusteePage(index : Int) extends QuestionPage[Boolean] {
```
Comment (adamconder) [inline app/pages/trustee/IsThisLeadTrusteePage.scala]: Remove this.

File: app/utils/AddATrusteeViewHelper.scala [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+          typeLabel = messages(s"entities.trustee.individual"),
+          changeUrl = "#",
+          removeUrl = controllers.trustee.routes.RemoveIndividualTrusteeController.onPageLoad(index).url
+        )
+      case (TrusteeType(None, Some(trusteeOrg)), index) =>
+        AddRow(
+          name = trusteeOrg.name,
+          typeLabel = messages(s"entities.trustee.organisation"),
+          changeUrl = "#",
+          removeUrl = controllers.trustee.routes.RemoveIndividualTrusteeController.onPageLoad(index).url
```
Comment (adamconder) [inline app/utils/AddATrusteeViewHelper.scala]: Check this?

File: test/models/StatusSpec.scala [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+package models
+
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalatest.prop.PropertyChecks
+import org.scalatest.{MustMatchers, OptionValues, WordSpec}
+import play.api.libs.json.{JsError, JsString, Json}
+
+class StatusSpec extends WordSpec with MustMatchers with PropertyChecks with OptionValues {
+
```
Comment (adamconder) [inline test/models/StatusSpec.scala]: Remove this.

File: app/utils/AddATrusteeViewHelper.scala (hunk 17 above)
Comment (adamconder) [reply app/utils/AddATrusteeViewHelper.scala]: @jassalrichy This will need to be it's own controller due to the change in the name structure for rendering

**Pull request merged**